vector/src/main/java/im/vector/activity/CommonActivityUtils.java

```diff
@@ -1684,6 +1685,50 @@ public static int getBadgeCount() {
         return mBadgeValue;
     }
 
+    /**
+     * Refresh the badge count when the device is offline.
+     * Notifications rooms are parsed to track the notification count value.
+     * @param aSession session value
+     * @param aActivity calling activity
+     */
+    public static void offlineRefreshBadgeUnreadCount(MXSession aSession, Activity aActivity) {
+        MXDataHandler dataHandler;
```

Comment:
i don't understand why an Activity is required here.
A context seems better.
For the linked f-droid issue, an activity won't be always available.